pkg/secrets/check_rights_windows.go

```diff
@@ -117,3 +115,150 @@ func checkRights(filename string, allowGroupExec bool) error {
 	}
 	return nil
 }
+
+func getACL(filename string) (*winutil.Acl, error) {
+	var fileDacl *winutil.Acl
+	err := winutil.GetNamedSecurityInfo(filename,
+		winutil.SE_FILE_OBJECT,
+		winutil.DACL_SECURITY_INFORMATION,
+		nil,
+		nil,
+		&fileDacl,
+		nil,
+		nil)
+
+	return fileDacl, err
+}
+
+func getLocalSystemSID() (*windows.SID, error) {
+	var localSystem *windows.SID
+	err := windows.AllocateAndInitializeSid(&windows.SECURITY_NT_AUTHORITY,
+		1, // local system has 1 valid subauth
+		windows.SECURITY_LOCAL_SYSTEM_RID,
+		0, 0, 0, 0, 0, 0, 0,
+		&localSystem)
+
+	return localSystem, err
+}
+
+func getAdministratorsSID() (*windows.SID, error) {
+	var administrators *windows.SID
+	err := windows.AllocateAndInitializeSid(&windows.SECURITY_NT_AUTHORITY,
+		2, // administrators group has 2 valid subauths
+		windows.SECURITY_BUILTIN_DOMAIN_RID,
+		windows.DOMAIN_ALIAS_RID_ADMINS,
+		0, 0, 0, 0, 0, 0,
+		&administrators)
+	return administrators, err
+}
+
+func getDDAgentUserSID() (*windows.SID, error) {
+	k, err := registry.OpenKey(registry.LOCAL_MACHINE, `SOFTWARE\Datadog\Datadog Agent`, registry.QUERY_VALUE)
+	if err != nil {
+		return nil, fmt.Errorf("could not open installer registry key: %s", err)
+	}
+	defer k.Close()
+
+	user, _, err := k.GetStringValue("installedUser")
+	if err != nil {
+		return nil, fmt.Errorf("could not read installedUser in registry: %s", err)
+	}
+
+	domain, _, err := k.GetStringValue("installedDomain")
+	if err != nil {
+		return nil, fmt.Errorf("could not read installedDomain in registry: %s", err)
+	}
+
+	sid, _, _, err := windows.LookupSID(domain, user)
+	return sid, err
+}
+
+var checkConfigRights = func(filename string) error {
+	if _, err := os.Stat(filename); err != nil {
+		if os.IsNotExist(err) {
+			return fmt.Errorf("config file %s does not exist", filename)
+		}
```

Comment:
If `os.Stat` hits errors, shouldn't we unconditionally jump out even if it's not a file-not-exist error?

Reply:
Good catch, this is a copy pasted bit, there is a similar issue in the `checkRights`. Fixing in both places.